### PR TITLE
[SOAR-15940] InsightIDR - advanced_query_on_log_set statistical result bug fix

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - to take the id that comes back from calling the `/query/logs` endpoint, then this ID can be used to the `/query/` endpoint to get back the correct values for the statistics, as the will not return the results we need

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

new unit test have been added and all of the old tests are passing 
```
Ran 82 tests in 0.038s

OK
```

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

testing via postman using a local docker container and showing the counts line up as expected 

image

image
cd

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/rapid7_insightidr/unit_test/test_advanced_query_on_log_set_unit.py
Comment (igorski-r7) [inline plugins/rapid7_insightidr/unit_test/test_advanced_query_on_log_set_unit.py:1]: Could you please add schema validation to those tests? 😄

**Review by igorski-r7 — APPROVED** [review]

File: plugins/rapid7_insightidr/help.md [hunk 1]
```diff
@@ -2024,7 +2024,8 @@ Example output:
 
 # Version History
 
-* 6.0.1 - Action: `Advanced Query On Log Set` - Up the maximium events returned from 50 to 500
+* 6.0.2 - Action: `Advanced Query On Log set` - Fixed error where statistical queries would always return 0.0
+* 6.0.1 - Action: `Advanced Query On Log` - Up the maximium events returned from 50 to 500
```
Comment (cmcnally-r7) [inline plugins/rapid7_insightidr/help.md:2028]: Just some grammar and spelling check 😄 
```suggestion
* 6.0.1 - Action: `Advanced Query On Log` - Increase the maximum results returned from 50 to 500
```

File: plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log_set/action.py [hunk 2]
```diff
@@ -173,7 +176,18 @@ def maybe_get_log_entries(
 
         results_object = response.json()
         if statistical:
-            potential_results = results_object.get("partial")
+            stats_endpoint = f"{self.connection.url}log_search/query/{results_object.get('id', '')}"
+            self.logger.info(f"Getting statistical from: {stats_endpoint}")
+            stats_response = self.connection.session.get(stats_endpoint, params=params)
+            try:
+                stats_response.raise_for_status()
+            except Exception:
```
Comment (cmcnally-r7) [inline plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log_set/action.py:184]: Exception is very broad here. Maybe narrow it down.
```suggestion
            except HTTPError as error:
```

File: plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log_set/action.py [hunk 3]
```diff
@@ -173,7 +176,18 @@ def maybe_get_log_entries(
...
+                raise PluginException(
+                    cause="Failed to get log sets from InsightIDR\n",
+                    assistance=f"Could not get statistical info from: {stats_endpoint}\n",
+                    data=stats_response.text,
```
Comment (cmcnally-r7) [inline plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log_set/action.py:188]:
```suggestion
                    data=f"{stats_response.text}, {error},
```

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

File: plugins/rapid7_insightidr/help.md (hunk 1 above)
Comment (rbowden-r7) [reply plugins/rapid7_insightidr/help.md:2028]: this has been updated

File: plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log_set/action.py (hunk 3 above)
Comment (rbowden-r7) [reply plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log_set/action.py]: The error message will also be raised here

**Review by cmcnally-r7 — APPROVED** [review]

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**